pkg/executor/push.go

```diff
@@ -115,6 +116,19 @@ func CheckPushPermissions(opts *config.KanikoOptions) error {
 			continue
 		}
 
+		// Historically kaniko was pre-configured by default with gcr credential helper,
+		// in here we keep the backwards compatibility by enabling the GCR helper only
+		// when gcr.io is in one of the destinations.
+		if strings.Contains(destRef.RegistryStr(), "gcr.io") {
+			// Checking for existence of docker.config as it's normally required for
+			// authenticated registries and prevent overwriting user provided docker conf
+			if _, err := os.Stat("/kaniko/.docker/config.json"); os.IsNotExist(err) {
```

Comment:
please define a constant for this file name in the same package.

Reply:
Done.